### PR TITLE
add a new gitaction to verify odh-model-controller image tag per branch

### PR DESCRIPTION
**Pull request by Jooho**
## Description

This is for https://issues.redhat.com/browse/RHOAIENG-9750
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by mwaykole — COMMENTED** [review]
/lgtm

File: .github/workflows/verify-odh-model-controller-img-tag.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    paths:
+      - "config/base/params.env"
+
+jobs:
+  check-image-tag:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (Jooho) [reply .github/workflows/verify-odh-model-controller-img-tag.yaml:15]: yep done.

File: .github/workflows/verify-odh-model-controller-img-tag.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          echo "PR_IMAGE_TAG=$IMAGE_TAG" >> $GITHUB_ENV
+
+      - name: Compare image tags
+        id: compare-tags
+        run: |
+          if [ "${{ github.base_ref }}" == "main" ]; then
+            EXPECTED_TAG="fast"
+          elif [[ "${{ github.base_ref }}" =~ ^release-(.*) ]]; then
+            VERSION=${BASH_REMATCH[1]}
+            EXPECTED_TAG="v$VERSION"
```
[the remark this answers is not in the document]
Comment (Jooho) [reply .github/workflows/verify-odh-model-controller-img-tag.yaml:30]: removed

File: .github/workflows/verify-odh-model-controller-img-tag.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          fi
+
+          # Check if PR_IMAGE_TAG matches EXPECTED_TAG followed by a dot and digits
+          PR_IMAGE_TAG=${{ env.PR_IMAGE_TAG }}          
+          if [[ ! "$PR_IMAGE_TAG" =~  ^"$EXPECTED_TAG"(\.[0-9]+)?$ ]]; then
+            echo "Image tag mismatch! Expected '$EXPECTED_TAG' but found '$PR_IMAGE_TAG'"
+            exit 1
+          fi
+
+          echo "Image tag('$PR_IMAGE_TAG') is correct."
```
[the remark this answers is not in the document]
Comment (Jooho) [reply .github/workflows/verify-odh-model-controller-img-tag.yaml:43]: updated.

File: .github/workflows/verify-odh-model-controller-img-tag.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          elif [[ "${{ github.base_ref }}" =~ ^release-(.*) ]]; then
+            VERSION=${BASH_REMATCH[1]}
+            EXPECTED_TAG="v$VERSION"            
+          else
+            echo "Unknown destination branch: ${{ github.base_ref }}"
+            exit 1
+          fi
+
+          # Check if PR_IMAGE_TAG matches EXPECTED_TAG followed by a dot and digits
+          PR_IMAGE_TAG=${{ env.PR_IMAGE_TAG }}
```
[the remark this answers is not in the document]
Comment (Jooho) [reply .github/workflows/verify-odh-model-controller-img-tag.yaml:37]: updated

**Pull request merged**